src/app.js

```diff
@@ -214,26 +214,28 @@ function openOrFocusTab(url) {
   if (appState.workTab === null) {
     app.makeNewWorkTab(url);
   } else {
-    app.refreshTabInfo();
+    app.checkAndOpenTab(url);
   }
 }
 
-// Make sure the work tab is open and in focus
-function refreshTabInfo() {
+// check to see if tab is valid to decide to open new tab or not
+
+function checkAndOpenTab(url) {
   chrome.tabs.get(appState.workTab.id, tab => {
     if (chrome.runtime.lastError) {
       appState.workTab = null;
+      app.makeNewWorkTab(url);
     } else {
       appState.workTab = tab;
-
-      // force selection
-      if (!appState.workTab.selected) {
-        chrome.tabs.update(appState.workTab.id, {selected: true});
+      const re = /tester\.rainforestqa\.com\/tester\//;  // test tab if worker is still working on the job
+      if (!re.test(tab.url)) {
+        app.makeNewWorkTab(url);
```

Comment:
!re.test(tab.url) is equal to re.test(tab.url) === false;    

this statement says,  if worker is not on the job url,  open a new tab.

normally your server doesn't response Yes, if worker is on the job.       This line just extra protection against if server response back yes when worker is still working on the job.       The job url is common at   https://tester.rainforest.com/tester/*    when they completed the job, it will change to https://tester.rainforest.com/thanks*

another reason for this..  is to protect the tab that have been reused again, if user entered a url on the same tab after completing the job...       

before this line, app.js doesn't have function to keep appState.workTab updated automatically if it's closed or reused.